### PR TITLE
Fixed duplicate mabar member in session (+ merging `main` to `dev` because stupid beastmaster didn't use the `dev` branch)

### PR DESCRIPTION
**Pull request by clawdius**
- Added check for **JoinGamingSession** function if the new member is already in the session. If it exist, then the message below will be shown.

image

_DB QoL suggestion:_
- Function like **GetMembersList** or acquiring document via **refId** should be in it's own file (**db_func.go** for example), so it can be reusable in other DB related function

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -95,14 +96,63 @@ func (d *DbClient) AddMemberToSession(ctx context.Context, refId string, newMemb
...
 
-	// Update member session
-	session.MembersSession = append(session.MembersSession, newMember)
-	_, err = docRef.Set(ctx, map[string]interface{}{
-		"members_sessions": session.MembersSession,
-	}, firestore.MergeAll)
+	// Get members, check if newMember already joined
+	members, _ := d.GetMembersList(ctx, refId)
+
+	if slices.Contains(members.MembersSession, newMember) {
```
Comment (bismastr) [inline db/db.go:102]: this is so sad, we combine business logic with db connection. we need to seperate this. but later

Comment (bismastr) [reply db/db.go:102]: can we put the logic inside handler fow now?

**Review by bismastr — CHANGES_REQUESTED** [review]
Seperate check if newMember already joined inside handler. 

Check isAlreadyInSeassion using refId from handler -> if yess addMember -> if no send msg component

**Comment by clawdius** [comment]
The `newMember` check now happens in the `action_session.go` handler.

Do we also need to make a dedicated `isAlreadyJoined` function in `action_session.go` handler? or we can just incorporate the `newMember` check in the `action_session.go`?

**Comment by clawdius** [comment]
Ok, now the dedicated `newMember` check happen in its own function (`IsInSession`), but the check is still happen in `JoinGamingSession`

**Pull request merged**